{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analiz adımları\n",
    "\n",
    "### Kalite kontrolü\n",
    "```\n",
    "[fastqc]\n",
    "tool = ${PIPELINE:fastqc}\n",
    "input_from = input_files\n",
    "args =  input_placeholder\n",
    "    -t 8\n",
    "    --outdir=${PROJECT:output_dir}\n",
    "dump_dir = fastqc\n",
    "```\n",
    "\n",
    "https://www.bioinformatics.babraham.ac.uk/projects/fastqc/\n",
    "\n",
    "komut:\n",
    "    fastqc Sample_R1.fq.gz\n",
    "\n",
    "\n",
    "### Hizalama/Haritalama (Alignment/Mapping) & Okumaların hazırlanması\n",
    "\n",
    "https://www.biostars.org/p/212136/\n",
    "    \n",
    "```\n",
    "[Fastq2Bam]\n",
    "tool = ${PIPELINE:bwa}\n",
    "sub_tool = mem\n",
    "input_from = input_files\n",
    "args = -t 8\n",
    "    -R \"@RG\\tID:name_placeholder\\tSM:name_placeholder\\tPL:illumina\\tLB:lib1\\tPU:foo\"\n",
    "    ${GENERAL:resource_dir}/${GENERAL:reference_genome}\n",
    "    input_placeholder |\n",
    "    samtools collate -@8 -O - - |\n",
    "    samtools fixmate -@8 -m -O bam - - |\n",
    "    samtools sort -@8 -O bam - |\n",
    "    samtools markdup -@8 -O bam - output_placeholder && samtools index -@8 output_placeholder\n",
    "named = True\n",
    "suffix = _bwa\n",
    "ext = bam\n",
    "input_multi = paired\n",
    "```\n",
    "\n",
    "#### BWA mem\n",
    "\n",
    "http://bio-bwa.sourceforge.net/bwa.shtml\n",
    "\n",
    "bwa referans_genome Sample_R1.fq.gz Sample_R2.fq.gz > Sample.sam\n",
    "\n",
    "#### Samtools collate\n",
    "komut:\n",
    "\n",
    "    samtools collate Sample.sam -o Sample_collate.bam\n",
    "\n",
    "amaç:\n",
    "    \n",
    "    Okumaları grupla.\n",
    "\n",
    "\n",
    "#### Samtools fixmate\n",
    "\n",
    "http://www.htslib.org/doc/samtools.html\n",
    "\n",
    "komut:\n",
    "\n",
    "    samtools fixmate Sample_collate.sam -o Sample_collate_fixmate.bam\n",
    "\n",
    "amaç:\n",
    "    \n",
    "    Eş okumaları işaretle markdup için.\n",
    "\n",
    "\n",
    "#### Samtools sort\n",
    "komut:\n",
    "    \n",
    "    samtools sort Sample_collate_fixmate.bam -o Sample_collate_fixmate_sort.bam\n",
    "amaç:\n",
    "    \n",
    "    Okumaları genomik pozisyonu göre sırala.\n",
    "\n",
    "#### Samtools markdup\n",
    "komut:\n",
    "    \n",
    "    samtools markdup Sample_collate_fixmate_sort.bam -o Sample_collate_fixmate_sort_markdup.bam\n",
    "amaç:\n",
    "    \n",
    "    Tekrar okumalarını işaretle.\n",
    "\n",
    "#### Samtools index\n",
    "komut:\n",
    "    \n",
    "    samtools index Sample_collate_fixmate_sort_markdup.bam\n",
    "amaç:\n",
    "    \n",
    "    bam dosyasını indexle.\n",
    "    \n",
    "\n",
    "### Varyant Çağırma\n",
    "```\n",
    "[deepvariant]\n",
    "container = ${PIPELINE:docker_deepvariant}\n",
    "tool = ${PIPELINE:deepvariant}\n",
    "input_from = input_files\n",
    "args =  --model_type=WES\n",
    "\t--ref=/resources/${GENERAL:reference_genome}\n",
    "\t--regions=\"/bed/${PROJECT:bed_file}\"\n",
    "\t--reads=/io/input_placeholder\n",
    "\t--output_vcf=/io/output_placeholder\n",
    "\t--num_shards=8\n",
    "\t&& ${PIPELINE:docker_deepvariant} sh -c \"grep '^#' /io/output_placeholder > /io/output_placeholder_ && grep -v '#' /io/output_placeholder | grep -v 'RefCall' >> /io/output_placeholder_ && mv -f /io/output_placeholder_ /io/output_placeholder\"\n",
    "ext = vcf\n",
    "output_dir = .\n",
    "input_dir = .\n",
    "```\n",
    "\n",
    "https://google.github.io/deepvariant/posts/2020-02-20-looking-through-deepvariants-eyes/\n",
    "\n",
    "komut:\n",
    "\n",
    "    deepvariant --model_type=WES --ref=referans_genome  --regions=bed_file --reads=Sample.bam --output_vcf=Sample.vcf\n",
    "\n",
    "### Anotasyon\n",
    "```\n",
    "[runVEPpick]\n",
    "container = ${PIPELINE:docker_vep}\n",
    "tool = ./vep\n",
    "input_from = input_files\n",
    "args =  --cache --refseq --everything\n",
    "\t# --offline\n",
    "\t--assembly GRCh37\n",
    "\t--flag_pick --pick_order ccds,tsl,rank,biotype,mane,appris,canonical,length\n",
    "\t--fork 8\n",
    "\t--format vcf --vcf --force_overwrite\n",
    "\t--fasta=/resources/${GENERAL:reference_genome}\n",
    "\t--dir_cache /opt/vep/.vep/ --dir_plugins /opt/vep/.vep/Plugins/\n",
    "\t--canonical\n",
    "\t--input_file /io/input_placeholder\n",
    "\t--output_file /io/output_placeholder\n",
    "\t--compress_output bgzip\n",
    "\t--plugin GO\n",
    "\t--plugin Phenotypes\n",
    "\t# --plugin Draw,output_placeholder,2000,100\n",
    "\t-custom /resources/hg19/clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNHGVS\n",
    "\t&& ${PIPELINE:docker_vep} rm /io/input_placeholder\n",
    "suffix = _vep\n",
    "ext = vcf.gz\n",
    "output_dir = .\n",
    "input_dir = .\n",
    "```\n",
    "\n",
    "https://www.ensembl.org/info/docs/tools/vep/script/vep_options.html\n",
    "\n",
    "komut:\n",
    "    ./vep --cache -i Sample.vcf -o Sample.vcf\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
